### PR TITLE
Add helper method to for IO connectors to use function secrets

### PR DESCRIPTION
**Pull request by jerrypeng**
### Motivation

Currently, we have begin tagging "sensitive" fields in source/sink configs.  We can leverage function secrets to deliver these sensitive configs to the function.

### Modifications

Add a helper method that devs can use to automatically inject "sensitive" configs passed in as secrets in to the sink/source config

I have modified the twitter firehose source to use the helper method as a demonstration of how to use it.

### CODE REVIEW

**Comment by jerrypeng** [comment]
@david-streamlio can you also take a look

**Comment by david-streamlio** [comment]
How would sensitive values such as passwords be passed in to the configuration?

**Comment by jerrypeng** [comment]
@david-streamlio taking the twitter source as an example:

```
name: test-source
tenant: public
namespace: default
topicName: persistent://public/default/jerry-output
archive: builtin://twitter
secrets:
  consumerKey:
    path: secret/data/twitter
    key: consumerKey
  consumerSecret:
    path: secret/data/twitter
    key: consumerSecret
  token:
    path: secret/data/twitter
    key: token
  tokenSecret:
    path: secret/data/twitter
    key: tokenSecret
```

File: pulsar-io/twitter/src/main/java/org/apache/pulsar/io/twitter/TwitterFireHose.java [hunk 1]
```diff
@@ -71,7 +72,7 @@
 
     @Override
     public void open(Map<String, Object> config, SourceContext sourceContext) throws IOException {
-        TwitterFireHoseConfig hoseConfig = TwitterFireHoseConfig.load(config);
+        TwitterFireHoseConfig hoseConfig = IOConfigUtils.loadWithSecrets(config, TwitterFireHoseConfig.class, sourceContext);
```
Comment (srkukarni) [inline pulsar-io/twitter/src/main/java/org/apache/pulsar/io/twitter/TwitterFireHose.java]: should this be gated with a config parameter? What if I'm just running a simple connector like twitter on my laptop without having any need for secrets?

Comment (jerrypeng) [reply pulsar-io/twitter/src/main/java/org/apache/pulsar/io/twitter/TwitterFireHose.java]: @srkukarni users can still pass the the "sensitive" fields as they have before.  This won't break that.

**Comment by jerrypeng** [comment]
rerun java8 tests

**Comment by jerrypeng** [comment]
rerun java8 tests

**Comment by jerrypeng** [comment]
rerun java8 tests

**Pull request merged**